modules/user_accounts/php/my_preferences.class.inc

```diff
@@ -0,0 +1,1134 @@
+                    if ($vals[1]==='1') {
+                        $defaults['examiner_radiologist'] = 'Y';
+                    } else {
+                        $defaults['examiner_radiologist'] = 'N';
+                    }
+                }
+                if ($vals[0]=='Y') {
+                    $defaults['ex_'.$cid] ='on';
+                }
+            }
+
+            // Notification defaults
+            // "notif_".$module."_".$operation."_".$service
+            $curr_sub = \NDB_Notifier::getUserNotificationModuleServices(
+                $user->getData("ID")
+            );
+            foreach ($curr_sub as $module=>$operations) {
+                foreach ($operations as $operation => $services) {
+                    unset($services['desc']);
+                    foreach ($services as $service => $subscribed) {
+                        $var_name = "notif_".$module."_".$operation."_".$service;
+
+                        if ($subscribed==='Y') {
+                            $defaults[$var_name] = 'on';
+                        }
+                    }
+                }
+            }
+        }
+        return $defaults;
+    }
+
+    /**
+     * Determines if the user is currently editing his/her own account.
+     *
+     * @return true if the user is editing his/her own account, false otherwise.
+     */
+    function _isEditingOwnAccount()
+    {
+        $editor = \User::singleton();
+        return !$this->isCreatingNewUser()
+        && ($editor->getUsername() == $this->identifier);
+    }
+
+    /**
+     * Processes the data entered in the form.
+     *
+     * @param array $values values entered in the form.
+     *
+     * @return void
+     */
+    function _process($values)
+    {
+        $config = \NDB_Config::singleton();
+        $DB     = \Database::singleton();
+
+        //The arrays that contain the edited permissions
+        $permissionsRemoved = array();
+        $permissionsAdded   = array();
+
+        $editor = \User::singleton();
+
+        // build the "real name"
+        $values['Real_name'] = $values['First_name'] . ' ' . $values['Last_name'];
+
+        //create the user
+        if (!is_null($this->identifier)) {
+            $user = \User::factory($this->identifier);
+        } else {
+            // Since the form has been validated there are two possibilities:
+            //    - UID is set
+            //    - UID is not set but the "Match UID to email" checkbox is checked
+            $effectiveUID = $values['NA_UserID'] == 'on'
+                ? $values['Email'] : $values['UserID'];
+            $user         = \User::factory($effectiveUID);
+        }
+
+        ////Get the current permissions/////
+        $current_permissionids = $user->getPermissionIDs();
+
+        $permIDs = array();
+        // store the permission IDs
+        if (!empty($values['permID'])) {
+            $permIDs = array_keys($values['permID']);
+        }
+        unset($values['permID']);
```

Comment:
all the stuff related to permissions should be taken out of the my_preferences page too.